### PR TITLE
[CSL-2185] Add dbgen to tools.

### PR DESCRIPTION
**Pull request by ksaric**
https://iohk.myjetbrains.com/youtrack/issue/CSL-2185

Adding https://github.com/adinapoli-iohk/dbgen to `cardano-sl` because of the GHC linker bug https://ghc.haskell.org/trac/ghc/ticket/13810

Example usage: `stack exec dbgen -- --config ./tools/src/dbgen/config.dhall --nodeDB db-mainnet --walletDB wdb-mainnet --configPath node/configuration.yaml --secretKey secret-mainnet.key --configProf mainnet_full`

Using current `mainnet.sh` script on `cardano-sl` root:
```
#!/usr/bin/env bash
set -eo pipefail

readonly CLUSTER=mainnet

if [[ "$1" == "-c" ]]; then
  shift
  rm -Rf                \
    db-${CLUSTER}       \
    wdb-${CLUSTER}      \
    secret-$CLUSTER.key \
    logs/$CLUSTER
fi

echo "Launch a single node and connect it to '${CLUSTER}' cluster..."

readonly TMP_TOPOLOGY_YAML=/tmp/topology.yaml
printf "wallet:
    relays: [[{ host: relays.cardano-mainnet.iohk.io }]]
    valency: 1
    fallbacks: 7" > "${TMP_TOPOLOGY_YAML}"

stack exec -- cardano-node                                  \
    --tlscert ./scripts/tls-files/server.crt                \
    --tlskey ./scripts/tls-files/server.key                 \
    --tlsca ./scripts/tls-files/ca.crt                      \
    --web                                                   \
    --no-ntp                                                \
    --topology "${TMP_TOPOLOGY_YAML}"                       \
    --log-config scripts/log-templates/log-config-qa.yaml   \
    --logs-prefix "logs/${CLUSTER}"                         \
    --db-path db-${CLUSTER}                                 \
    --wallet-db-path wdb-${CLUSTER}                         \
    --keyfile secret-$CLUSTER.key                           \
    --configuration-key mainnet_full
```

### CODE REVIEW

File: tools/src/dbgen/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+In order for the program to work correctly, it must be provided with a non-empty RocksDB generated
+with the same configuration this program is invoked with, or the DB-content deserialisation will
+fail with the "wrong magic" error.
+
+## Usage
+
+First compile `dbgen` against the same revision of Cardano you want it to generate the `wallet-db` for.
+At the moment the program has been tested with `release/1.0.4`
+(SHA: `6e53bf599097aa0b55738a454115f76f69c9489e`). Then edit the `config.dhall` file to specify the number
```
Comment (adinapoli-iohk) [inline tools/src/dbgen/README.md]: This section of the readme is stale, but can be changed later.

File: tools/src/dbgen/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+- Faking `StakeLock` sync to avoid the relevant `MVar` from deadlocking.
+
+- By default, the current version of the wallet creates with each new wallet a default account and a default
+  address.
+
+## To implement
+
+- @Ilya Peresadin - probably we also should add option for `dbgen` which generates ton of history entries (https://input-output-rnd.slack.com/archives/C7DRYGPT8/p1515507443000578)
+
+- @Ilya Peresadin and @George Agapov - add fake utxo entries to local db (https://input-output-rnd.slack.com/archives/C7DRYGPT8/p1515493184000042)
```
Comment (adinapoli-iohk) [inline tools/src/dbgen/README.md]: Probably better to move these into tickets rather than as parts of the README.

Comment (dcoutts) [reply tools/src/dbgen/README.md]: Yes, at least need tickets.

**Comment by adinapoli-iohk** [comment]
@dcoutts This is a PR which targets `master` (as it targets the remedial work) so unless you think we should not bother and open it against `develop`, just you have the power to merge it 😁

**Pull request merged**